### ISSUE TITLE
Are there limitations on the bandwidth for bridged connections?

### ISSUE THREAD

**Issue by RalphCodesTheInternet** [garywill/linux-router#55]
**Are there limitations on the bandwidth for bridged connections?**

**How to reproduce:**
On my Pi I have two WiFi adaptors, both supporting AP. Wlan0 and Wlan1 respectively.
One adaptor is used to connect to an available access point that supplies internet. The second adaptor is used to create the access point on the Pi which I can connect to with other devices. Which one is used does not appear to make a difference.

So by default the connection is shared across adaptors. The problem however is that it appears that the connection speed when bridged hovers around the 10mbps range.

**Tests ive done:**
Ive used multiple different WiFi adapters, in different combination. With the same outcome of 10mbps. (Adaptors with greater output power. Some with multiple Tx and Rx antennas)
Ive connected one devices at a time with the same outcome.
Ive connected about 10 devices all with the same outcome simultaneously.  
Ive downloaded the speed test cli client and tested all the adaptors I have. Connected them to an available access point and tested the speed coming into the Pi. All the adaptors perform at 100% the speed of my ISP supplied connection. 

So basically the problem is when broadcasting the connection across another WiFi adaptor, it appears to be reduced in the range of 70% - 80%. Is this expected in such a setup? Is there some kind of throttling implemented somewhere? 

#1 This issue also mentions something about the speed.

Thanks in advance

**Comment by kphanipavan** [comment]
Hello,

I am experiencing the same 10-15Mbps cap on internet bandwidth with my setup.

tl;dr: try using ingress(wifi-internet) and egress(hotspot) interfaces from the same device/wifi adaptors, it fixed the bandwidth cap.

My setup:

Device: Radxa Rock 5B SBC, Rockchip RK3588 SoC
Wireless NICs: Radxa A8 PCIe NIC(Realtek RTL8852be), TP-Link Archer T3U USB NIC(Realtek RTL8812bu)
Wired NICs: RTL8125 2.5GbE Builtin
OS: Ubuntu 22.04, Hostapd built from source

I wanted to use the A8 module as the hotspot interface as it supports WiFi6, 5GHz and at 80MHz. I was able to get 80MHz on 5GHz channel working using lnxrouter, but I was facing issue of limited bandwidth to the broader internet. The ingress wifi connection is my Univ's 5G APs, which peak at 200Mbps. And as stated, when connected to my hotspot, I get 15Mbps peak, 9Mbps continuous for download.

I then tried wired network input (laptop to SBC wired internet share using lnxrouter) and hotspot over A8, which resulted in the same slow speeds, both to local connections(SMB file transfer from laptop to phone connected to hotspot) and to the internet. The local file transfer and ping were sub optimal. I then tried using the A8 module for both connecting to wifi and for hotspot, which fixed both ping spikes and slow bandwidth.

So my guess would be that, as you said at the end, bridging internet between 2 different adaptors yields slow internet bandwidth.

I tried setting the Nice and IO Nice values but that didn't change much. The bottleneck maybe RAM Speed and/or CPU, as my RK3588 SoC is a lot faster than your RPi's, but I get the same 10M. This is the only info I have.

~Pvn

**Comment by kphanipavan** [comment]
Hello,

Can you/anyone suffering from bandwidth issues try removing any power saving tags in ht_capab or vht_capab? See if that improves the bandwidth

I removed `[SMPS-STATIC]` in my config and I am getting max bandwidth, albeit when using the same card for Wifi and Hotspot.

The card may get hot bcz of no power saving, will report my findings...

~Pvn
